pkg/services/live/features/broadcast.go

```diff
@@ -1,32 +1,86 @@
 package features
 
 import (
+	"encoding/json"
+
 	"github.com/centrifugal/centrifuge"
 	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/util"
 )
 
 // BroadcastRunner will simply broadcast all events to `grafana/broadcast/*` channels
-// This makes no assumptions about the shape of the data and will broadcast it to anyone listening
-type BroadcastRunner struct{}
+// This assumes that data is a JSON object
+type BroadcastRunner struct {
+	Publisher models.ChannelPublisher
+}
 
 // GetHandlerForPath called on init
-func (g *BroadcastRunner) GetHandlerForPath(path string) (models.ChannelHandler, error) {
-	return g, nil // for now all channels share config
+func (b *BroadcastRunner) GetHandlerForPath(path string) (models.ChannelHandler, error) {
+	return b, nil // for now all channels share config
+}
+
+// DoNamespaceHTTP called from the http api
+func (b *BroadcastRunner) DoNamespaceHTTP(c *models.ReqContext) {
+	c.JSON(400, util.DynMap{
+		"Unsupportedd": "BroadcastRunner",
+	})
 }
 
 // GetChannelOptions called fast and often
-func (g *BroadcastRunner) GetChannelOptions(id string) centrifuge.ChannelOptions {
-	return centrifuge.ChannelOptions{}
+func (b *BroadcastRunner) GetChannelOptions(id string) centrifuge.ChannelOptions {
+	return centrifuge.ChannelOptions{
+		// HistorySize:    1,
+		// HistoryRecover: true,
+	}
 }
 
 // OnSubscribe for now allows anyone to subscribe to any dashboard
-func (g *BroadcastRunner) OnSubscribe(c *centrifuge.Client, e centrifuge.SubscribeEvent) error {
+func (b *BroadcastRunner) OnSubscribe(c *centrifuge.Client, e centrifuge.SubscribeEvent) error {
 	// anyone can subscribe
 	return nil
 }
 
 // OnPublish called when an event is received from the websocket
-func (g *BroadcastRunner) OnPublish(c *centrifuge.Client, e centrifuge.PublishEvent) ([]byte, error) {
+func (b *BroadcastRunner) OnPublish(c *centrifuge.Client, e centrifuge.PublishEvent) ([]byte, error) {
 	// expect the data to be the right shape?
 	return e.Data, nil
 }
+
+// DoChannelHTTP accepts POST from anywhere
+func (b *BroadcastRunner) DoChannelHTTP(c *models.ReqContext, channel string) {
+	if c.Req.Method == "POST" {
```

Comment:
It's better/more idiomatic to invert the `if` here and fail and return if the method isn't "POST". It will make the code more readable due to reduced nesting.